libminifi/src/sitetosite/RawSocketProtocol.cpp

```diff
@@ -240,6 +240,12 @@ bool RawSiteToSiteClient::initiateCodecResourceNegotiation() {
   }
 }
 
+namespace {
+void logPortStateError(std::shared_ptr<core::logging::Logger>& logger, utils::Identifier port_id, const std::string& error) {
```

Comment:
It's a dilemma to me. If it's a lambda, it has to be stored in a variable, that is allocated every time the function runs. Not a big performance issue BTW.

Reply:
@adam-markovics that's going to be inlined and optimized out, no variable is created on the stack, unless you're using `-O0`.
My concern is that it's more complex than the raw calls, but it's shorter as well, and it's all local, so it doesn't really matter. I'm neutral.

Reply:
I trust the compiler and since its slightly more readable I've changed it (thanks for the suggestion) :+1: 
https://github.com/apache/nifi-minifi-cpp/pull/1314/commits/deb192a736ab05fc942379c959d97a6657d9aae0